tools/validate_spec.py

```diff
@@ -79,81 +78,134 @@ def cli(argv: List[str]) -> Union[int, str]:
+        message_path,
+        original_message,
+        valid_original_message,
+        valid_parser_result,
+    )
 
-    message_name = str(identifier.name)
-    package_name = str(identifier.parent)
-    try:
-        pdu_message = PyRFLX.from_specs([str(path_spec)], skip_model_verification=no_verification)[
-            package_name
-        ][message_name]
-    except KeyError as e:
-        raise ValidationError(
-            f'message "{message_name}" could not be found in package "{package_name}"'
-        ) from e
-    except FileNotFoundError as e:
-        raise ValidationError(f"specification {e}") from e
-
-    classified_incorrectly = 0
-    with JsonOutputWriter(full_output_path) as json_output_writer:
-        validator = Validator(pdu_message)
-        for path, path_generator in path_generators.items():
-            for message_file_path in path_generator:
-                msg = get_message_to_validate(message_file_path, path == path_valid)
-                validation_result = validator.validate_message(msg)
-                json_output_writer.write(validation_result)
-                if validation_result.classification in [
-                    Classification.FP,
-                    Classification.FN,
-                ]:
-                    print(f"{message_file_path}: FAILED")
-                    print(f"{validation_result.get_abbreviated_output()}\n")
-                    if abort_on_error:
-                        raise ValidationError(
-                            f"{message_file_path} classified as "
-                            f"{validation_result.classification.value}"
-                        )
-                    classified_incorrectly += 1
-                else:
-                    print(f"{message_file_path}: PASSED")
-
-    if classified_incorrectly != 0:
-        raise ValidationError(f"{classified_incorrectly} messages were classified incorrectly")
-
-
-class JsonOutputWriter:
+
+@dataclass
+class ValidationResult:
+    validation_success: bool
+    parser_result: Optional[MessageValue]
+    parser_error: Optional[str]
+    message_file_path: Path
+    original_message: bytes
+    valid_original_message: bool
+    valid_parser_result: bool
+
+    def __get_parser_result_field_values(self) -> Dict[str, object]:
+        if self.parser_result is None:
+            return {}
+        parsed_field_values: Dict[str, object] = {}
+        for field_name in self.parser_result.fields:
+            try:
+                field_value = self.parser_result.get(field_name)
+                if isinstance(field_value, MessageValue):
+                    field_value = field_value.bytestring.hex()
+                if isinstance(field_value, bytes):
+                    field_value = field_value.hex()
+                parsed_field_values[field_name] = field_value
+            except PyRFLXError as e:
+                parsed_field_values[field_name] = f"{e} or not set"
+        return parsed_field_values
```

Comment:
It should be possible to differentiate invalid fields and not set fields.

Reply:
The cleanest way to do this might be to add a method/property to the `MessageValue` class to query the status of a field (`set` or `unset`) or get a list of all set fields. The `MessageValue.Field` class specifically has a `set` attribute, but there is currently no way to access this information outside of the `MessageValue` class.

I don't see a good solution using `required_fields` and/or `accessible fields`.

Reply:
cf. https://github.com/Componolit/RecordFlux/pull/616#issuecomment-806541975

Reply:
How would that solve the problem? We would still just get a `PyRFLXError` here. The description in an exception is just for debugging purposes, the control flow should never depend on the description of an exception.

Reply:
I think it is not possible to have a field with an invalid value in the first place. Attempting to assign an invalid value should trigger an exception while parsing or setting the field. In the validation tool, this would mean that `parser_result` is `None` and the `__get_parser_result_field_values` method would never be called. 
As the method is only executed in case `parser_result` is not `None` (thus if the message has been parsed successfully), the only possibility where a field is not valid (and therefore an exception is generated) is, when it is not set.

Reply:
The information I would find helpful in case of an error is a list of all the fields which were correctly parsed and the field which made the parser fail. For the former the value of the fields should be given, for the latter the error message. I think listing all other fields is unnecessary.

Reply:
Like this?
```
"file name": "tests/data/ethernet/frame/invalid/ethernet_invalid_too_short.raw",
	"provided as": true,
	"recognized as": false,
	"original": "17f7a643be6530fa4457452543358bdad0807bbc26604f19a0cbec84e5353806daf3e81939675d2d287eae6f5ad799700c06290fddaf31d621",
	"parsed field values": {
		"Destination": 26352413818469,
		"Source": 53851446527269,
		"Type_Length_TPID": 17205
	},
	"error": "pyrflx: error: cannot set value for field Ether_Type\npyrflx: error: Number 17205 is not a valid enum value"
```
It is not possible to have `Ether Type` in the `parsed field values` list. If an exception is raised the value is not set to the field. The only indication where the error occurred is in the error message itself.

Reply:
The format looks fine to me.

Reply:
Added in 4c047b6